tests/runner/test_storage.py

```diff
@@ -481,3 +484,101 @@ def test_to_pull_step_with_plaintext(self):
                 ),
             ):
                 repo.to_pull_step()
+
+
+class TestRemoteStorage:
+    def test_init(self):
+        rs = RemoteStorage("s3://bucket/path")
+        assert rs._url == "s3://bucket/path"
+        assert rs.pull_interval == 60
+
+    def test_get_required_package_for_scheme(self):
+        assert RemoteStorage._get_required_package_for_scheme("s3") == "s3fs"
+        assert RemoteStorage._get_required_package_for_scheme("gs") == "gcsfs"
+        assert RemoteStorage._get_required_package_for_scheme("unknown") is None
+
+    def test_filesystem(self, monkeypatch):
+        mock_filesystem = MagicMock()
+        monkeypatch.setattr("fsspec.filesystem", mock_filesystem)
+
+        key = Secret(value="fake")
+        secret = Secret(value="fake")
+        token = Secret(value="fake")
+
+        rs = RemoteStorage("s3://bucket/path", key=key, secret=secret, token=token)
+        rs._filesystem
+        mock_filesystem.assert_called_once_with(
+            "s3", key="fake", secret="fake", token="fake"
+        )
+
+    def test_set_base_path(self):
+        rs = RemoteStorage("s3://bucket/path")
+        path = Path.cwd() / "new_base_path"
+        rs.set_base_path(path)
+        assert rs._storage_base_path == path
+
+    def test_destination(self):
+        rs = RemoteStorage("s3://bucket/path")
+        assert rs.destination == Path.cwd() / Path("bucket") / Path("path")
+
+    @pytest.mark.asyncio
+    async def test_pull_code(self, monkeypatch):
```

Comment:
I may have missed it but was there a test for when this fails?

Reply:
Nope, added one in [14e2f60](https://github.com/PrefectHQ/prefect/pull/11072/commits/14e2f6074b76b81bd80be5be66874bd3ebfa3cd8)